### DISCUSSION TITLE
nested state

### DISCUSSION THREAD

**Discussion by krisrajzlive** [mobxjs/mobx#2507] [Q&A] [answered]
I am looking out for help on updating nested state without flattening them out every time. It is so annoying if there are more than two levels of nesting. Thanks. Here's the state object:

`searchData:
    {
        results:
        {
            response:'',
            isAvailable: false,
        },
        tasks:
        {
            pageIndex: 0,
            currentPage: 0,
            totalPages: 0,
            recordsPerPage: 10,
            totalRecords: 0
        },
        projects:
        {
            pageIndex: 0,
            currentPage: 0,
            totalPages: 0,
            recordsPerPage: 10,
            totalRecords: 0
        },
        criteria:
        {
            validationMessage: "",
            searchText: "",
            projectSite: "",
            regions: ["all"],
            practices: ["all"],
            offices: ["all"],
            statuses: ["all"]
        }
     }`

**Comment by mweststrate** [comment, marked as answer]
MobX doesn't require any flattening, and updating nested state should be as simple as `a.b.c = d`

Beyond that, if you post questions please include some example code so that others have a vague idea what you are talking about. We can't see your screen so your question is extremely vague as it stands and not likely to receive any useful answer :)
